### ISSUE TITLE
Reading energy dispersion in MapDataset.read()

### ISSUE THREAD

**Issue by AtreyeeS** [gammapy/gammapy#2513]
It seems that there is some conflict between the `MapDataset.from_hdulist()` and `MapData.to_hdulist()` methods.

```
data_store = DataStore.from_dir("$GAMMAPY_DATA/cta-1dc/index/gps/")
observations = data_store.get_observations([110380])

energy_axis = MapAxis.from_edges(
    np.logspace(-1.0, 1.0, 5), unit="TeV", name="energy", interp="log"
)
geom = WcsGeom.create(width=(4, 4), axes=[energy_axis], coordsys="GAL")

stacked = MapDataset.create(geom)
for obs in observations:
    maker = MapDatasetMaker(geom, offset_max=4.0 * u.deg)
    dataset = maker.run(obs)
    stacked.stack(dataset)

filename = "test.fits.gz"
stacked.write(filename)
stacked1 = MapDataset.read(filename)
```

`stacked1.edsip` and `stacked1.psf` is `None` (whereas `stacked.edsip` and `stacked.psf`) exist.

This seems to be because of mismatch in the hdu names between the `from_hdulist()` and `to_hdu_list()` methods.

**Comment by adonath** [comment]
Thanks @AtreyeeS! Indeed we're not reading the the `EdispMap` and `PSFMap` in `MapDataset.from_hdulist()` correctly. We should extend `.from_hdulist()` to do this and add a regression test.

**Comment by AtreyeeS** [comment]
Addressed in #2523.
